x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_reassign_policy_modal/index.tsx

```diff
@@ -146,7 +135,10 @@ export const AgentReassignAgentPolicyModal: React.FunctionComponent<Props> = ({
             <EuiSelect
               fullWidth
               isLoading={agentPoliciesRequest.isLoading}
-              options={policySelectOptions}
+              options={agentPolicies.map((agentPolicy) => ({
```

Comment:
I prefer to keep this modal pretty simple, even if suboptimal. There are only 2 interactions in the UI that would cause a re-render: the user selects a new policy or they click the confirm button. Neither are going to be high frequency actions that creating this small function again is going to cause a problem.

I think premature optimization is what caused this bug in the first place 🙃

Reply:
Even if we extract the function, it'll still be recreated on each render. The only way to preserve a function's identity across renders is with `useMemo/useCallback`, which feels heavy-handed here.